### PR TITLE
Improved performance of finding ongoing challenges;

### PR DESCRIPTION
**Pull request by spherefoundry**
# Checklist:

- [x] Code follows the style guidelines
- [x] Documentation updated
- [x] Corner cases evaluated and handled
- [x] Automatic tests added/update
- [x] Automatic tests pass
- [x] Software runs (locally or hosted)
- [x] Sanity check passed
- [x] Self-review passed

### CODE REVIEW

File: src/services/challenges_repository.js [hunk 1]
```diff
@@ -50,14 +61,19 @@ export default class ChallengesRepository {
...
+    const filtered = this.filterOutFinishedChallenges(
+      startedChallenges,
+      resolvedChallenges,
+      timedOutChallenges,
+      this.challengesWrapper.defaultAddress
+    );
+    const sorted = this.sortChallenges(
+      filtered
     );
+    return sorted;
```
Comment (spherefoundry) [inline src/services/challenges_repository.js]: This is a little bit artificial (we could just do a straight return), but makes the code more readable, and adding performance measure points easier.

File: src/services/challenges_repository.js [hunk 2]
```diff
@@ -17,23 +16,35 @@ export default class ChallengesRepository {
...
+          acc[challengeId] -= 1;
         }
-        return put(acc, challengeId, acc[challengeId] - 1);
+        return acc;
       },
       startedCount
     );
-    return allChallenges.filter(({challengeId}) => unresolvedCount[challengeId] > 0 && !timedOutSet.has(challengeId));
+    const filtered = allChallenges.filter(({challengeId}) => unresolvedCount[challengeId] > 0 && !timedOutSet.has(challengeId));
+    return filtered;
```
Comment (krzysztof-jelski) [inline src/services/challenges_repository.js]: I would inline `filtered`, but I presume it's just as with `sorted` elsewhere - done to make adding performance measurement easier.

Comment (spherefoundry) [reply src/services/challenges_repository.js]: 👍
